src/main/java/com/iwaconsolti/league/manager/service/impl/SoccerLeagueImpl.java

```diff
@@ -0,0 +1,180 @@
+package com.iwaconsolti.league.manager.service.impl;
+
+import com.iwaconsolti.league.manager.model.Matches;
+import com.iwaconsolti.league.manager.model.Players;
+import com.iwaconsolti.league.manager.model.Teams;
+import com.iwaconsolti.league.manager.service.ILeagues;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service("soccerLeague")
+@Slf4j
+public class SoccerLeagueImpl implements ILeagues {
+
+    @Value("${league.teams.limit}")
```

Comment:
i will suggest to always add a default value to avoid any problem if the settings is missing int he properties